octodns/provider/ns1.py

```diff
@@ -275,49 +545,342 @@ def populate(self, zone, target=False, lenient=False):
+                        'meta': {key: [value]},
+                    },
                 )
-        self.log.debug("params for A: %s", params)
-        return params
+
+        params['filters'] = []
+        if has_country:
+            params['filters'].append(
+                {"filter": "shuffle", "config": {}}
+            )
+            params['filters'].append(
+                {"filter": "geotarget_country", "config": {}}
+            )
+            params['filters'].append(
+                {"filter": "select_first_n",
+                 "config": {"N": 1}}
+            )
+
+        return params, None
+
+    def _monitors_for(self, record):
+        monitors = {}
+
+        if getattr(record, 'dynamic', False):
+            expected_host = record.fqdn[:-1]
+            expected_type = record._type
+
+            for monitor in self._client.monitors.values():
+                data = self._parse_notes(monitor['notes'])
+                if expected_host == data['host'] and \
+                   expected_type == data['type']:
+                    # This monitor does not belong to this record
+                    config = monitor['config']
+                    value = config['host']
+                    monitors[value] = monitor
+
+        return monitors
+
+    def _uuid(self):
+        return uuid4().hex
+
+    def _feed_create(self, monitor):
+        monitor_id = monitor['id']
+        self.log.debug('_feed_create: monitor=%s', monitor_id)
+        # TODO: looks like length limit is 64 char
+        name = '{} - {}'.format(monitor['name'], self._uuid()[:6])
+
+        # Create the data feed
+        config = {
+            'jobid': monitor_id,
+        }
+        feed = self._client.datafeed_create(self._client.datasource_id, name,
+                                            config)
+        feed_id = feed['id']
+        self.log.debug('_feed_create:   feed=%s', feed_id)
+
+        return feed_id
+
+    def _monitor_create(self, monitor):
+        self.log.debug('_monitor_create: monitor="%s"', monitor['name'])
+        # Create the notify list
+        notify_list = [{
+            'config': {
+                'sourceid': self._client.datasource_id,
+            },
+            'type': 'datafeed',
+        }]
+        nl = self._client.notifylists_create(name=monitor['name'],
+                                             notify_list=notify_list)
+        nl_id = nl['id']
+        self.log.debug('_monitor_create:   notify_list=%s', nl_id)
+
+        # Create the monitor
+        monitor['notify_list'] = nl_id
+        monitor = self._client.monitors_create(**monitor)
+        monitor_id = monitor['id']
+        self.log.debug('_monitor_create:   monitor=%s', monitor_id)
+
+        return monitor_id, self._feed_create(monitor)
+
+    def _monitor_gen(self, record, value):
+        host = record.fqdn[:-1]
+        _type = record._type
+
+        request = 'GET {path} HTTP/1.0\\r\\nHost: {host}\\r\\n' \
```

Comment:
Optional, but can also be written like this I believe
```suggestion
        request = r'GET {path} HTTP/1.0\r\nHost: {host}\r\n' \
```